common/src/main/java/com/wynntils/gui/screens/maps/MainMapScreen.java

```diff
@@ -213,33 +212,30 @@ public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTic
     }
 
     private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
-        List<Poi> pois = new ArrayList<>();
+        Stream<? extends Poi> pois = Models.Map.getServicePois().stream();
 
-        pois.addAll(Models.Map.getServicePois());
-        pois.addAll(Models.Map.getCombatPois());
-        pois.addAll(Models.Map.getLabelPois());
+        pois = Stream.concat(pois, Models.Map.getCombatPois().stream());
+        pois = Stream.concat(pois, Models.Map.getLabelPois().stream());
+        pois = Stream.concat(pois, MapFeature.INSTANCE.customPois.stream());
 
-        pois.addAll(MapFeature.INSTANCE.customPois);
+        // Make sure compass and player pois are on top
```

Comment:
@kristofbolyai Is this still a relevant/correct comment? I thought the order of Pois were determined by their Z (or Y or whatever) position in this branch?

Reply:
The comment probably stayed after I removed sorting

Reply:
Ok if I remove it?

Reply:
Yes, thanks

Reply:
I did anyway :)